DataFormats/Detectors/FIT/FT0/src/RecPoints.cxx

```diff
@@ -29,24 +31,36 @@ void RecPoints::fillFromDigits(const o2::ft0::Digit& digit)
   constexpr Int_t nMCPs = nMCPsA + nMCPsC;
   Float_t sideAtime = 0, sideCtime = 0;
 
-  std::vector<o2::ft0::ChannelData> chDgDataArr;
+  //  mIntRecord = digit.getInteractionRecord();
+  mTimeStamp = o2::InteractionRecord::bc2ns(mIntRecord.bc, mIntRecord.orbit);
+
+  LOG(INFO) << " event time " << mTimeStamp << " orbit " << mIntRecord.orbit << " bc " << mIntRecord.bc;
+
+  int nbc = ft0BCData.size();
+    LOG(INFO) << "Entry " << ient << " : " << nbc << " BCs stored";
+    int itrig = 0;
 
-  mIntRecord = digit.getInteractionRecord();
-  mEventTime = o2::InteractionRecord::bc2ns(mIntRecord.bc, mIntRecord.orbit);
-  LOG(INFO) << " event time " << mEventTime << " orbit " << mIntRecord.orbit << " bc " << mIntRecord.bc;
-  mTimeAmp = digit.getChDgData();
-  for (auto& d : mTimeAmp) {
-    d.CFDTime *= 13;
-    d.QTCAmpl /= Geometry::MV_2_Nchannels;
-    LOG(DEBUG) << " mcp " << d.ChId << " cfd " << d.CFDTime << " amp " << d.QTCAmpl;
-    chDgDataArr.emplace_back(o2::ft0::ChannelData{d.ChId, int(d.CFDTime), int(d.QTCAmpl), d.numberOfParticles});
-    setChDgData(std::move(chDgDataArr));
-    if (std::fabs(d.CFDTime) < 2000) {
-      if (d.ChId < nMCPsA) {
-        sideAtime += d.CFDTime;
+    for (int ibc = 0; ibc < nbc; ibc++) {
+      const auto& bcd = ft0BCData[ibc];
+      if (bcd.triggers >0) {
+        LOG(INFO) << "Triggered BC " << itrig++;
+      }
+      bcd.print();
+      //
+      auto channels = bcd.getBunchChannelData(ft0ChData);
+      int nch = channels.size();
+      for (int ich = 0; ich < nch; ich++) {
+        channels[ich].CFDTime *= 13;
+        channels[ich].QTCAmpl /= Geometry::MV_2_Nchannels;
```

Comment:
could you please define in the Geometry the inverse of MV_2_Nchannels and multiply by it?